### PR TITLE
JSON renderer

### PR DESCRIPTION
**Pull request by iddan**
Issues #44

### CODE REVIEW

File: pyinstrument/renderers.py [hunk 1]
```diff
@@ -128,6 +129,21 @@ def render_frame(self, frame):
...
+        'line_no': frame.line_no,
+        'time': frame.time(),
+        'children': [_json_frame(frame) for frame in frame.children]
+    }
+
+class JSONRenderer(Renderer):
+    preferred_recorder = 'time_aggregating'
+
+    def render(self, frame):
+        return json.dumps(_json_frame(frame))
```
[the remark this answers is not in the document]
Comment (iddan) [reply pyinstrument/renderers.py]: I though about it: many outputs can generate big outputs so adding spacing chars would increase their size significantly. Maybe this should be another renderer / CLI option?
If you go the UNIX way you can always do: `python -m pyinstrument script.py | python -m json.tool`

File: pyinstrument/renderers.py [hunk 2]
```diff
@@ -128,6 +129,21 @@ def render_frame(self, frame):
         result += '</div></div>'
 
         return result
+            
+def _json_frame(frame):
```
[the remark this answers is not in the document]
Comment (iddan) [reply pyinstrument/renderers.py]: No problem

File: pyinstrument/renderers.py [hunk 3]
```diff
@@ -128,6 +129,21 @@ def render_frame(self, frame):
...
+    return {
+        'function': frame.function,
+        'file_path': frame.file_path_short,
```
[the remark this answers is not in the document]
Comment (iddan) [reply pyinstrument/renderers.py]: Cool

**Comment by iddan** [comment]
Here is the app expecting the result: https://python-flame-chart.netlify.com

File: pyinstrument/renderers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joerick) [reply pyinstrument/renderers.py]: Hmm. I don't agree. JSON is a really verbose format. All those repeated key names! All those numbers written out as strings. 

The big advantage of JSON is human-readability - you can look at it and quickly understand it. To strip indentation is to lose that. 

If size becomes a meaningful issue the right answer would be to remove frames with tiny durations, or use something like Protobuf.

(Re. piping through `python -m json.tool` - it's a good idea, but I don't think the burden of a good user experience should be on the user)

[the remark this answers is not in the document]
Comment (iddan) [reply pyinstrument/renderers.py]: Okay, will add

**Comment by joerick** [comment]
Just waiting on the `indent=2` change and then I can merge this! :)

**Comment by iddan** [comment]
Done!

**Comment by joerick** [comment]
Cheers @iddan !

**Comment by iddan** [comment]
Please let me know when it is released

**Comment by joerick** [comment]
Apologies for the delay - working on a few more changes and tests before the next release. In the meantime, you can install from the master branch directly - `pip install https://github.com/joerick/pyinstrument/archive/master.zip`

**Comment by joerick** [comment]
Released as 2.2.1!

**Pull request merged**